main.py

```diff
@@ -0,0 +1,84 @@
+from openpyxl import load_workbook
+from textwrap import dedent
+
+
+class Node:
+    def __init__(self, label, right_child, left_child, predicate, fields):
+        try:
+            self.label = int(label)
+        except ValueError:
+            self.label = label
+        except TypeError:
+            self.label = "empty"
+
+        try:
+            self.right_child = int(right_child)
+        except ValueError:
+            self.right_child = f"\"{right_child}\""
+        except TypeError:
+            self.right_child = "empty"
+
+        try:
+            self.left_child = int(left_child)
+        except ValueError:
+            self.left_child = f"\"{left_child}\""
+        except TypeError:
+            self.left_child = "empty"
+
+        self.predicate = dedent(predicate) if predicate else ""
+        self.fields = str.join(" ", dedent(fields).splitlines()) if fields else ""
+
+    def __str__(self):
+        result = dedent(f"""
+        @tree.node({self.label}, {self.right_child}, {self.left_child})
+        def node_{self.label}(clue: ClaimLineFocus, registry: Registry, cue: ClaimFocus) -> bool:
```

Comment:
I would cut it down to just `clue: ClaimLineFocus`
```suggestion
        def node_{self.label}(clue: ClaimLineFocus) -> bool:
```